src/main/java/org/mockbukkit/mockbukkit/block/data/encoder/PropertyEncoder.java

```diff
@@ -0,0 +1,24 @@
+package org.mockbukkit.mockbukkit.block.data.encoder;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+
+@ApiStatus.Internal
+public interface PropertyEncoder<T>
+{
+
+	@Nullable
+	Object encode(@Nullable T value);
```

Comment:
Why not this?
```suggestion
	String encode(@Nullable T value);
```
I don't understand why a non string value would be needed here

Reply:
I changed from String to object, because of this:
<img width="385" height="376" alt="imagem" src="https://github.com/user-attachments/assets/015f535d-f382-4b22-8d5c-b94764f084ae" />
There are some objects that are not string, like the case of boolean